Fix: Update networkmanager.md to address #702 (and some other bits of cleanup on that page whilst I'm at it)

11BelowStudio opened the pull request:
**Select the type of change:**
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Small Changes - Typos, formatting, slight revisions
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site and Docusaurus


**Purpose of the Pull Request:**

* Lines 67-72: Fixed the `UnityTransport` connection data stuff (see #702)
* Line 29: formatting (inline code snippet is now code formatted)
* Line 79: Explicitly mentioning that `Shutdown()` disconnects self (if client) or all clients (if server).
* Lines 107-117: Added a `NetworkManager.IsServer` check (if not server, it will complain and return without doing anything), to make it clear that disconnecting clients is server-only.
* Lines 143-208:
  * `Awake()`: Destroys the component if singleton exists already (so there's not multiple of them lying around)
  * `OnDestroy()` Checks if the component being destroyed is the singleton, only unassigning the singleton if this is the singleton. Also removing this component's callbacks from the NetworkManager singleton (cleanup yaaay)


**Issue Number:** 
<!-- Post the issue or ticket number addressed by the PR. -->
#702

NoelStephensUnity commented:
@11BelowStudio 
I added a small portion of the code related suggested changes in this PR into another PR that further clarifies a few things regarding the example code snippets as well as the ConnectionNotificationManager in this PR:
https://github.com/Unity-Technologies/com.unity.multiplayer.docs/pull/705

The text copy of this PR I really want to keep, but before I moved forward if you wouldn't mind looking over the changes to the code examples in the above PR and see if those adjustments make sense?  (I really like feedback and it helps greatly to have as many "eyes" on it as possible...super appreciated!)

11BelowStudio commented:
> @11BelowStudio I added a small portion of the code related suggested changes in this PR into another PR that further clarifies a few things regarding the example code snippets as well as the ConnectionNotificationManager in this PR: #705
> 
> The text copy of this PR I really want to keep, but before I moved forward if you wouldn't mind looking over the changes to the code examples in the above PR and see if those adjustments make sense? (I really like feedback and it helps greatly to have as many "eyes" on it as possible...super appreciated!)

Looking over the changes to the code examples now, you're still doing the connection data stuff incorrectly, currently editing the version I have I can open a PR with some working code.

11BelowStudio commented:
@NoelStephensUnity see #708
